platform-sdk/docs/proposals/001-tss-rosters-design-proposal/proposal.md

```diff
@@ -18,61 +18,86 @@ components within the platform.
+
+To manage `Roster`s in the state, a new set of `store` APIs will be introduced, similar to existing store
+implementations. A `ReadableRosterStore` and `WritableRosterStore` will be introduced.
+
+The `ReadableRosterStore` will have methods such as `getCandidateRoster()`, `getActiveRoster()` and `getRosterHistory()`.
+These will respectively return the candidate `Roster`, active `Roster`, and the history of active rosters present in the state.
+
+A `WritableRosterStore` implementation will have methods to store `roster`s in the state such as `setCandidateRoster()` and `setActiveRoster()`.
+
+The `WritableRosterStore#setCandidateRoster()` method will be used to set a candidate `Roster` in the state as follows:
+
+1. Validate the candidate `Roster`.
+2. Store the candidate `Roster` hash in the `RosterState`, and the Roster itself in the `RosterMap`.
+
+The `WritableRosterStore#setActiveRoster()` method will be called to set a new active `Roster` in the state. A call to `setActiveRoster` will infer the following operations:
+
+1. Invalidate and remove any existing candidate `Roster` present in the state
+2. Rotate the current active `Roster` to the previous active `Roster`
+3. Set the new Roster as the active `Roster` in the state
 
 ```java
-//in PlaformState
+//in WritableRosterStore
 
 void setCandidateRoster(@NonNull final Roster candidateRoster);
+void setActiveRoster(@NonNull final Roster roster, final long round);
 ```
 
-The steps to validate and add the Roster to the state will be executed within the same thread that submits the request.
-An exception will be thrown if the Roster is invalid. The durability of the submitted Roster is ensured via storing
-it in the state.
+```java
+//in ReadableRosterStore
 
-The existing `PlatformState` class is in the middle of a refactor, and in the long term, may cease to exist. However,
-its replacement will continue to have this new API to set a candidate roster.
+@Nullable
+Roster getCandidateRoster();
+Roster getActiveRoster();
+```
+
+The steps to validate and add the Roster to the state will be executed within the same thread that submits the request.
+A caught exception will be thrown if the Roster is invalid. The durability of the submitted Roster is ensured by storing it in the state.
 
 ## Data Structure
 
-The App is already responsible for managing the address book. We propose that it continues to do so, and when it
-deems necessary (such as when it receives a HAPI transaction), creates a candidate Roster object and passes it to
-the platform via the new API. The platform will then validate the roster and store it in the state.
+The App is already responsible for managing the address book. We propose that it continues to do so, and when it deems necessary (such as during a freeze upgrade), creates a candidate `Roster` object and passes it to the platform via the new API.
+The platform will then validate the roster and store it in the state.
```

Comment:
This looks a bit inaccurate. The Roster Service is an app/services-level entity. It has nothing to do with the "platform". Further, the "platform" (whatever that means nowdays) doesn't even need to know about candidate rosters. It will be told what Roster to use as an active roster (and what prior history of rosters is) on startup.

Reply:
Thanks. I've further revised the use of the word "platform".